org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/ResourceLeakTests.java

```diff
@@ -7269,4 +7269,26 @@ public void testGH2129() {
 		"",
 		options);
 }
+public void testGH2642() {
+	Map options = getCompilerOptions();
+	options.put(CompilerOptions.OPTION_ReportPotentiallyUnclosedCloseable, CompilerOptions.ERROR);
+	options.put(CompilerOptions.OPTION_ReportUnclosedCloseable, CompilerOptions.ERROR);
+	options.put(CompilerOptions.OPTION_ReportExplicitlyClosedAutoCloseable, CompilerOptions.ERROR);
+	runLeakTest(
+		new String[] {
+			"DemoNonCloseableWarning.java",
+			"""
+			import java.io.FileWriter;
+			public class DemoNonCloseableWarning {
+			    public static void main(String[] args) throws Exception {
+			        try (FileWriter writer = new FileWriter("/dev/null")) {
```

Comment:
@stephan-herrmann : this doesn't work on Windows, sorry for providing this example.

We have new test fails: https://download.eclipse.org/eclipse/downloads/drops4/I20240716-1800/testresults/html/org.eclipse.jdt.core.tests.compiler_ep433I-unit-win32-java17_win32.win32.x86_64_17.html

Reply:
I don't have a windows box to test what works on windows.
Do you have a better example, or must I add a dummy error to make this a negative test not trying to execute the program?

Reply:
Why do we execute the snippet at all? Easiest is not to run it.

Reply:
Another possibility is to use some other "writer" like StringWriter (haven't checked if that would show same issue, but I assume it should).

Reply:
> Why do we execute the snippet at all? Easiest is not to run it.

We have two strategies:
* runNegativeTest() look for expected compile errors
* runConformTest() expect no compiler errors but then run the program and expect correct output from it

Adding a new parameter to runConformTest() would need to trickle down a myriad of delegating overloads, not a nice option.

So that standard trick to avoid executing is to use runNegativeTest() - with a dummy error like a field with type Zork :)